ios/ReactNativePageView.m

```diff
@@ -10,412 +9,333 @@
-            scrollView.contentOffset = CGPointMake(scrollView.bounds.size.width, 0);
+    if (!self.overdrag) {
+        if (self.orientation == UIPageViewControllerNavigationOrientationHorizontal) {
+            if (self.currentPage == 0 && scrollView.contentOffset.x < scrollView.bounds.size.width) {
+                scrollView.contentOffset = CGPointMake(scrollView.bounds.size.width, 0);
+            } else if (self.currentPage == self.count - 1 && scrollView.contentOffset.x > scrollView.bounds.size.width) {
+                scrollView.contentOffset = CGPointMake(scrollView.bounds.size.width, 0);
+            }
+        } else {
+            if (self.currentPage == 0 && scrollView.contentOffset.y < scrollView.bounds.size.height) {
+                scrollView.contentOffset = CGPointMake(0, scrollView.bounds.size.height);
+            } else if (self.currentPage == self.count - 1 && scrollView.contentOffset.y > scrollView.bounds.size.height) {
+                scrollView.contentOffset = CGPointMake(0, scrollView.bounds.size.height);
+            }
         }
     }
-    
-    if (self.isHorizontal) {
+
+    if (!self.onPageScroll) {
+        return;
+    }
+    CGPoint point = scrollView.contentOffset;
+    float offset = 0;
+    if (self.orientation == UIPageViewControllerNavigationOrientationHorizontal) {
         if (self.frame.size.width != 0) {
-            offset = (point.x - self.frame.size.width)/self.frame.size.width;
+            offset = (point.x - self.frame.size.width) / self.frame.size.width;
         }
     } else {
         if (self.frame.size.height != 0) {
-            offset = (point.y - self.frame.size.height)/self.frame.size.height;
+            offset = (point.y - self.frame.size.height) / self.frame.size.height;
         }
     }
-
-    float absoluteOffset = fabs(offset);
-    if(absoluteOffset > 1) {
-        absoluteOffset = 1.0;
+    if (fabs(offset) > 1) {
+        offset = offset > 0 ? 1.0 : -1.0;
     }
-    
-    NSString *scrollDirection = [self determineScrollDirection:scrollView];
-    NSString *oppositeDirection = self.isHorizontal ? @"left" : @"up";
-    NSInteger position = self.currentIndex;
-
-    if(absoluteOffset > 0) {
-        position = [scrollDirection  isEqual: oppositeDirection] ? self.currentIndex - 1 : self.currentIndex;
-        absoluteOffset =  [scrollDirection  isEqual: oppositeDirection] ? 1 - absoluteOffset : absoluteOffset;
+    NSInteger position = self.currentPage;
+    if (offset < 0 && position > 0) {
+        offset += 1;
+        position -= 1;
     }
-   
-    
-    self.lastContentOffset = scrollView.contentOffset;
-    [self.eventDispatcher sendEvent:[[RCTOnPageScrollEvent alloc] initWithReactTag:self.reactTag position:@(position) offset:@(absoluteOffset)]];
+    self.onPageScroll(@{
+        @"offset": [NSNumber numberWithFloat:offset],
+        @"position": [NSNumber numberWithInteger:position]
+    });
 }
 
-- (NSString *)determineScrollDirection:(UIScrollView *)scrollView {
```

Comment:
Why it has been removed ?

Reply:
The function is not used by any other code.